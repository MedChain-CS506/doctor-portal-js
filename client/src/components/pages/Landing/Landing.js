import React from 'react';

import { Link } from 'react-router-dom';

import PropTypes from 'prop-types';

import { makeStyles, Fab, Typography } from '@material-ui/core';
import GitHubIcon from '@material-ui/icons/GitHub';
import FavoriteIcon from '@material-ui/icons/Favorite';
import AddCircleIcon from '@material-ui/icons/AddCircle';
import Search from '../../layout/Search';

const useStyles = makeStyles(theme => ({
  center: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    textAlign: 'center',
  },

  button: {
    marginTop: theme.spacing(1),
  },

  buttonIcon: {
    marginRight: theme.spacing(1),
  },

  patientFormIcon: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2),
    background: 'red',
    color: 'white',
  },
}));

const Landing = ({ signedIn = false, isPharmacist }) => {
  const classes = useStyles();

  if (signedIn && !isPharmacist) {
    return (
      <>
        <Link to="/patient-form">
          <Fab
            data-testid="new-patient-button"
            className={classes.patientFormIcon}
            variant="extended"
          >
            <AddCircleIcon className={classes.buttonIcon} /> New Patient Form
          </Fab>
        </Link>
        <Search isPharmacist={isPharmacist} />
      </>
    );
  }

  if (isPharmacist) {
    return <Search isPharmacist={isPharmacist} />;
  }

  return (
    <div className={classes.center}>
      <FavoriteIcon color="action" />
      <Typography color="textSecondary" variant="h3">
        {process.env.REACT_APP_NAME}
      </Typography>
      <Typography data-testid="basic-desc" color="textSecondary" variant="subtitle1">
        The simplest decentralized medical-records application
      </Typography>
      <Fab
        data-testid="github-link"
        className={classes.button}
        color="secondary"
        href="https://github.com/MedChain-CS506"
        rel="noopener noreferrer"
        target="_blank"
        variant="extended"
      >
        <GitHubIcon className={classes.buttonIcon} /> Repo
      </Fab>
    </div>
  );
};

Landing.propTypes = {
  signedIn: PropTypes.bool.isRequired,
  isPharmacist: PropTypes.bool,
};

export default Landing;
