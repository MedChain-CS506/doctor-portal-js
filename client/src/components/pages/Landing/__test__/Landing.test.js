import React from "react";

import ReactDOM from "react-dom";
import { BrowserRouter as Router } from "react-router-dom";

import Landing from "../Landing";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <Router>
      <Landing signedIn={false} />
    </Router>,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
});

it("renders correctly with signedIn being false", () => {
  const { getByTestId } = render(
    <Router>
      <Landing signedIn={false} />
    </Router>
  );
  expect(getByTestId("basic-desc")).toHaveTextContent(
    "The simplest decentralized medical-records application"
  );
});

it("renders correctly with signedIn being true, as a doctor", () => {
  const { getByTestId } = render(
    <Router>
      <Landing signedIn={true} isPharmacist={false} />
    </Router>
  );
  // expect(getByTestId("search-bar")).toHaveTextContent(
  //   process.env.REACT_APP_NAME
  // );
  expect(getByTestId("add-patient-button")).toBeTruthy();
  expect(getByTestId("search-bar")).toBeTruthy();
  expect(getByTestId("search-patient-form")).toBeTruthy();
});
