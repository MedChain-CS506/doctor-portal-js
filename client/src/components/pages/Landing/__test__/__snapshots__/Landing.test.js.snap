// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with signedIn being true, as a pharmacist 1`] = `[Function]`;
